packages/components/src/Lettera.purs

```diff
@@ -76,15 +80,57 @@ letteraSearchUrl = letteraBaseUrl <> "/list/search"
 letteraAdvertorialUrl :: String
 letteraAdvertorialUrl = letteraBaseUrl <> "/list/active-advertorial"
 
+encodeRequestHeaders :: Array AX.RequestHeader -> Maybe Json
+encodeRequestHeaders [] = Nothing
+encodeRequestHeaders hs =
+  let encodeHeader h obj = flip extend obj $ case h of
+        AX.Accept (MediaType v)      -> "Accept" := v
+        AX.ContentType (MediaType v) -> "ContentType" := v
+        AX.RequestHeader k v         -> k := v
+  in Just $ foldr encodeHeader jsonEmptyObject hs
+
+encodeRequest :: forall a . AX.Request a -> Json
+encodeRequest x =
+  let method = case x.method of
+        Left m  -> encodeJson (show m)
+        Right s -> encodeJson (show s)
+  in "method"  := method ~>
+     "url"     := x.url  ~>
+     "headers" :=? encodeRequestHeaders x.headers ~>?
+     -- Currently we're only using GET requests so we don't need content:
+     -- "content" := x.content ~>
+     jsonEmptyObject
+
+encodeResponseHeaders :: Array AX.ResponseHeader -> Json
+encodeResponseHeaders hs =
+  let encodeHeader (AX.ResponseHeader k v) obj = k := v ~> obj
+  in foldr encodeHeader jsonEmptyObject hs
+
+encodeResponse :: forall a . AX.Response a -> Json
+encodeResponse x =
+  let statusCode = case x.status of
+        StatusCode n -> n
+  in "statusCode" := statusCode ~>
+--     "statusText" := x.statusText ~>
+     "headers" := encodeResponseHeaders x.headers ~>
+     jsonEmptyObject
+
+encodeError :: String -> Maybe String -> Maybe String -> Json
+encodeError name req err =
+  "error" := name ~>
+  (("request" := _) <$> req) ~>?
+  (("message" := _) <$> err) ~>?
+  jsonEmptyObject
+
 data LetteraError
-  = ResponseError AX.Error
-  | HttpError Int
+  = ResponseError String AX.Error
+  | HttpError String String
```

Comment:
Status code is used in downstream, not just for logging, so it needs to stay as `Int`.

Reply:
Ok, now we have two `Json` values as well as one `Int`.